packages/@o3r/schematics/src/utility/wrapper.ts

```diff
@@ -7,14 +14,58 @@ const noopSchematicWrapper: SchematicWrapper = (fn) => fn;
  * if @o3r/telemetry is installed
  * @param schematicFn
  */
-export const createSchematicWithMetricsIfInstalled: SchematicWrapper = (schematicFn) => (opts) => async (_, context) => {
+export const createSchematicWithMetricsIfInstalled: SchematicWrapper = (schematicFn) => (opts) => async (tree, context) => {
   let wrapper: SchematicWrapper = noopSchematicWrapper;
+  const packageJsonPath = '/angular.json';
+  const packageJson = tree.exists(packageJsonPath) ? tree.readJson(packageJsonPath) as JsonObject : {};
   try {
     const { createSchematicWithMetrics } = await import('@o3r/telemetry');
-    wrapper = createSchematicWithMetrics;
+    if ((packageJson.cli as JsonObject)?.o3rMetrics) {
+      wrapper = createSchematicWithMetrics;
+    }
   } catch (e: any) {
     // Do not throw if `@o3r/telemetry is not installed
-    context.logger.debug('`@o3r/telemetry` is not available\n' + e.toString());
+    if ((packageJson.cli as JsonObject)?.o3rMetrics === true) {
+      context.logger.info('`cli.o3rMetrics` is set to true in your package.json, please install the telemetry package with `ng add @o3r/telemetry` to enable the collect of metrics.');
```

Comment:
Sound more like a warning :)

```suggestion
      context.logger.warn('`cli.o3rMetrics` is set to true in your package.json, please install the telemetry package with `ng add @o3r/telemetry` to enable the collect of metrics.');
```